Mathlib/Order/Filter/Pi.lean

```diff
@@ -131,10 +131,27 @@ theorem pi_principal [Finite ι] (s : (i : ι) → Set (α i)) :
     pi (fun i ↦ 𝓟 (s i)) = 𝓟 (univ.pi s) := by
   simp [Filter.pi, Set.pi_def]
 
+theorem mem_pi_principal {t : Set ((i : ι) → α i)} :
+    t ∈ pi (fun i ↦ 𝓟 (s i)) ↔ ∃ I : Set ι, I.Finite ∧ I.pi s ⊆ t :=
+  (hasBasis_pi (fun i ↦ hasBasis_principal _)).mem_iff.trans <| by simp
+
+theorem hasBasis_pi_principal (s : (i : ι) → Set (α i)) :
+    HasBasis (pi fun i ↦ 𝓟 (s i)) Set.Finite (Set.pi · s) :=
+  ⟨fun _ ↦ mem_pi_principal⟩
+
 @[simp]
 theorem pi_pure [Finite ι] (f : (i : ι) → α i) : pi (pure <| f ·) = pure f := by
   simp only [← principal_singleton, pi_principal, univ_pi_singleton]
 
+theorem mem_pi_pure {f : (i : ι) → α i} {s : Set ((i : ι) → α i)} :
```

Comment:
This seems like it would be undesirable for the case when `Finite ι` _is_ true. Perhaps the docstring should point out `pi_pure` as a more convenient lemma for that case?

They are of course right next to each other, but that wouldn't be true for someone reading a hover docstring.

Reply:
@eric-wieser Added docstrings.